core/async/src/test_loop/futures.rs

```diff
@@ -1,16 +1,11 @@
-use std::{
-    fmt::Debug,
-    sync::{Arc, Mutex},
-    task::Context,
-    time::Duration,
-};
-
-use futures::{
-    channel::oneshot,
-    future::BoxFuture,
-    task::{waker_ref, ArcWake},
-    FutureExt,
-};
+use futures::channel::oneshot;
+use futures::future::BoxFuture;
```

Comment:
I didn't notice it earlier, but prefer tokio::sync::oneshot (and in general, treat tokio as more reliable than other crates implementing async primitives - at least that's what matklad told me).